hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/ThrottleAccumulator.java

```diff
@@ -16,35 +16,553 @@
 
-    /**
-     * Tests whether the given query should be throttled, assuming its functionality is as
-     * specified.
+        gasThrottle.leakUnusedGasPreviouslyReserved(value);
+    }
+
+    /*
+     * Gets the current list of active throttles.
      *
-     * @param functionality the functionality of the query
-     * @param query the query to test
-     * @return true if the query should be throttled, false otherwise
+     * @return the current list of active throttles
+     */
+    @NonNull
+    public List<DeterministicThrottle> allActiveThrottles() {
+        return activeThrottles;
+    }
+
+    /*
+     * Gets the current list of active throttles for the given functionality.
+     *
+     * @param function the functionality to get the active throttles for
+     * @return the current list of active throttles for the given functionality
+     */
+    @NonNull
+    public List<DeterministicThrottle> activeThrottlesFor(@NonNull final HederaFunctionality function) {
+        final var manager = functionReqs.get(function);
+        if (manager == null) {
+            return Collections.emptyList();
+        } else {
+            return manager.managedThrottles();
+        }
+    }
+
+    /*
+     * Indicates whether the last transaction was throttled by gas.
+     *
+     * @return whether the last transaction was throttled by gas
+     */
+    public boolean wasLastTxnGasThrottled() {
+        return lastTxnWasGasThrottled;
+    }
+
+    /*
+     * Checks if the given functionality should be throttled by gas.
+     *
+     * @param function the functionality to check
+     * @return whether the given functionality should be throttled by gas
+     */
+    public static boolean isGasThrottled(@NonNull final HederaFunctionality function) {
+        return GAS_THROTTLED_FUNCTIONS.contains(function);
+    }
+
+    private boolean shouldThrottleTxn(
+            @NonNull final TransactionInfo txnInfo, @NonNull final Instant now, @NonNull final HederaState state) {
+        final var function = txnInfo.functionality();
+        final var configuration = configProvider.getConfiguration();
+
+        // Note that by payer exempt from throttling we mean just that those transactions will not be throttled,
+        // such payer accounts neither impact the throttles nor are they impacted by them
+        // In the current mono-service implementation we have the same behavior, additionally it is
+        // possible that transaction can also be exempt from affecting congestion levels separate from throttle
+        // exemption
+        // but this is only possible for the case of triggered transactions which is not yet implemented (see
+        // MonoMultiplierSources.java)
+        final var isPayerThrottleExempt = throttleExempt(txnInfo.payerID(), configuration);
+        if (isPayerThrottleExempt) {
+            return false;
+        }
+
+        // TODO: in mono we pass: () -> getSpanMapAccessor().getEthTxDataMeta(this) as 3rd param to
```

Comment:
That's a performance optimization; the behavior here should be functionally identical.